subsys/testsuite/include/ztest.ld

```diff
@@ -1,28 +0,0 @@
-/*
- * Copyright 2021 Google LLC
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-SECTIONS
-{
-	.data.ztest_suite_node_area : ALIGN(4)
```

Comment:
I think you still need to copy this line over.

Reply:
I tried that just now but when I run `readelf` on the binary I was seeing that the sections were placed in the wrong location. I've update the PR just now with something that builds at least and is cleaner (again the 3rd commit is the one I'm trying to debug).

Reply:
@dcpleung `if (suite->before) {` appears to be the offending instruction but I'm not seeing why it would do so. I added more details to the discord chat thread

Reply:
I got a bit further by using `RAM_SECTIONS` instead of the data sections

Reply:
Updated again, more boards now pass, but I'm still seeing the same error in other boards. It seems to believe that the data in the test suite is not acceissble.